src/sonic-config-engine/config_samples.py

```diff
@@ -47,49 +48,105 @@ def generate_empty_config(data):
         new_data['DEVICE_METADATA']['localhost']['type'] = 'LeafRouter'
     return new_data
 
+def port_is_uplink(port_info):
+    return len(port_info['lanes'].split(',')) == 4
+
+def port_is_downlink(port_info):
+    return len(port_info['lanes'].split(',')) == 2
+
+def get_uplinks_downlinks(port_table):
+    # sort the ports based on the first lane number
+    sorted_ports = sorted(port_table.items(), key=lambda x: int(x[1]['lanes'].split(',')[0]))
+    uplinks = []
+    downlinks = []
+    ports_iter = iter(sorted_ports)
+    for port_tuple in ports_iter:
+        # Iterate over ports two at a time so we can get
+        # used ports and the corresponding unused ports
+        port_a_name, port_a_info = port_tuple
+        port_b_name, port_b_info = next(ports_iter)
+        if port_is_uplink(port_a_info) and port_is_uplink(port_b_info) :
+            uplinks.append(port_a_name)
+            uplinks.append(port_b_name)
+        elif port_is_downlink(port_a_info) and port_is_downlink(port_b_info):
+            downlinks.append(port_a_name)
+            # don't use port_b in this case
+        # this shouldn't happen
+        else:
```

Comment:
Lets not raise exception here. It could be possible to have single lane ports like [here](https://github.com/Azure/sonic-buildimage/blob/master/device/arista/x86_64-arista_7050cx3_32s/Arista-7050CX3-32S-C32/port_config.ini#L34).

Reply:
method is removed in the latest update